common/src/HttpContentExtensions.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.IO;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Devices.Shared
+{
+    internal static class HttpContentExtensions
+    {
+        internal static async Task CopyToStreamAsync(this HttpContent content, Stream stream, CancellationToken cancellationToken)
+        {
+#if NET5_0
+            await content.CopyToAsync(stream, cancellationToken).ConfigureAwait(false);
+#else
+            _ = cancellationToken;
```

Comment:
I wouldn't skimp on code comments here. Just even saying .NET 5.0 includes a cancellation token parameter, which we'd prefer to make use of. For the other framework versions, we'll just need to ignore the parameter.

Reply:
sure